services/queue/src/utils.js

```diff
@@ -0,0 +1,133 @@
+const assert = require('assert').strict;
+const { fromNow } = require('taskcluster-client');
+const { paginateResults } = require('taskcluster-lib-api');
+const debug = require('debug')('utils');
+
+const artifactUtils = {
+  // Create a single instance, or undefined, from a set of rows containing zero
+  // or one elements.
+  fromDbRows(rows) {
+    if (rows.length === 1) {
+      return exports.artifactUtils.fromDb(rows[0]);
+    }
+  },
+  // Create a single instance from a DB row
+  fromDb(row) {
+    return {
+      taskId: row.task_id,
+      runId: row.run_id,
+      name: row.name,
+      storageType: row.storage_type,
+      contentType: row.content_type,
+      details: row.details,
+      present: row.present,
+      expires: row.expires,
+      etag: row.etag,
+    };
+  },
+  // Create a serializable representation of this namespace suitable for response
+  // from an API method.
+  serialize(artifact) {
+    return {
+      storageType: artifact.storageType,
+      name: artifact.name,
+      expires: artifact.expires.toJSON(),
+      contentType: artifact.contentType,
+    };
+  },
+  //  Get all expired artifacts using a handler function.
+  // This is to avoid loading all epxired queue artifacts in memory.
+  async getExpiredArtifacts(db, expires, handler) {
+    assert(!handler || handler instanceof Function,
+      'If options.handler is given it must be a function');
+
+    const fetchResults = async (continuation) => {
+      const query = continuation ? { continuationToken: continuation } : {};
+      const {continuationToken, rows} = await paginateResults({
```

Comment:
paginateResults is for APIs.. it's just abstracting over the pagination in a way that obscures what's really happening here.  I think this function could just loop with adjustments to `page_offset` on each iteration, until it gets zero results.